src/Compiler/Checking/MethodCalls.fs

```diff
@@ -279,16 +279,32 @@ let rec AdjustRequiredTypeForTypeDirectedConversions (infoReader: InfoReader) ad
 
     // Adhoc int32 --> int64
     elif g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions && typeEquiv g g.int64_ty reqdTy && typeEquiv g g.int32_ty actualTy then 
-       g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
+        g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
 
     // Adhoc int32 --> nativeint
     elif g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions && typeEquiv g g.nativeint_ty reqdTy && typeEquiv g g.int32_ty actualTy then 
-       g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
+        g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
 
     // Adhoc int32 --> float64
     elif g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions && typeEquiv g g.float_ty reqdTy && typeEquiv g g.int32_ty actualTy then 
-       g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
+        g.int32_ty, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), None
 
+    elif g.langVersion.SupportsFeature LanguageFeature.NullableOptionalInterop && isMethodArg && isNullableTy g reqdTy && not (isNullableTy g actualTy) then 
+        let reqdTy2 =
+            if isTyparTy g reqdTy then
+                let tp = destTyparTy g reqdTy 
+                match tp.Constraints |> List.choose (function TyparConstraint.CoercesTo (c, _) -> Some c | _ -> None) with
+                | [reqdTy2] when tp.Rigidity = TyparRigidity.Flexible -> reqdTy2
+                | _ -> reqdTy
+            else reqdTy
+       
+        if not (isTyparTy g reqdTy2) &&
+            (not (TypeFeasiblySubsumesType 0 g infoReader.amap m reqdTy2 CanCoerce actualTy) ||
+            isTyparTy g actualTy && (let ftyvs = freeInType CollectAll reqdTy2 in ftyvs.FreeTypars.Contains(destTyparTy g actualTy))) then
+
+            actualTy, TypeDirectedConversionUsed.Yes(warn TypeDirectedConversion.BuiltIn), Some (reqdTy, reqdTy2, ignore)
```

Comment:
Regardless of missed simplification, this bit of code worries me. It's written in such a way it mirrors logic in TryFindRelevantImplicitConversion but originally NullableInterop did not ever return any new equational constraints. Maybe eqn here should always be None?

Reply:
Ok I've removed it.

This was a holdover from earlier code when this arm was not protected by isMethodArg, where it would have needed to support the same solving complexity as the op_Implicit arm.